### PR TITLE
Feature/#8 dataloader refactoring

### PR DESCRIPTION
**Pull request by mrTahion**
Внесены изменения в загрузку данных, небольшие улучшения cada-vae процедуры.

### CODE REVIEW

File: zeroshoteval/utils/checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Args:
+        model (model): model to save the weight to the checkpoint.
+        data (dict): dictionary with embeddings dataset and extra data.
+        cfg (CfgNode): configs to save.
+    """
+    # Ensure that the save dir exists.
+    PathManager.mkdirs(get_embeddings_dir(path_to_job))
+
+    # Record the state.
+    data["cfg"] = {"cfg": cfg.dump()}
```
Comment (termit209) [inline zeroshoteval/utils/checkpoint.py]: data["cfg"] = {"cfg": cfg.dump()} почему так а не  data["cfg"] = cfg.dump()

File: zeroshoteval/utils/checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+def get_path_to_data(path_to_job, cfg):
+    """
+    Get the full path to a data file.
+
+    Args:
+        path_to_job (string): the path to the folder of the current job.
+        cfg(CfgNode): configs. Details can be found in
+            zeroshoteval/config/defaults.py
+    """
+    name = f"{cfg.ZSL_MODEL_NAME}_{cfg.DATA.DATASET_NAME}_emb.pyth"
```
Comment (termit209) [inline zeroshoteval/utils/checkpoint.py:32]: что за расширение .path?

File: zeroshoteval/dataset/dataset.py [hunk 3]
```diff
@@ -77,98 +79,42 @@ def __getitem__(self, idx):
...
-            att_splits["trainval_loc"].squeeze() - 1
-        )  # --> train_feature = TRAIN SEEN
+        self.train_indices = att_splits["trainval_loc"].squeeze() - 1
         self.test_seen_indices = att_splits["test_seen_loc"].squeeze() - 1
         self.test_unseen_indices = att_splits["test_unseen_loc"].squeeze() - 1
-        self.test_indices = np.append(
-            self.test_seen_indices, self.test_unseen_indices
-        )
+        self.test_indices = None
+        self.la = labels
```
Comment (termit209) [inline zeroshoteval/dataset/dataset.py]: почему не оставить self.labels , а не self.la как раньше, разве так не будет понятнее?

**Review by nikitashvarts — APPROVED** [review]

File: zeroshoteval/dataset/dataset.py (hunk 3 above)
Comment (mrTahion) [reply zeroshoteval/dataset/dataset.py]: удалил лишнее

File: zeroshoteval/utils/checkpoint.py (hunk 1 above)
Comment (mrTahion) [reply zeroshoteval/utils/checkpoint.py]: поправил

File: zeroshoteval/utils/checkpoint.py (hunk 2 above)
Comment (mrTahion) [reply zeroshoteval/utils/checkpoint.py:32]: *`.pyth`

Как я понимаю нет жесткого требования по расширениям, ряд библиотек использует `.pyth` для сохранения чекпоинтов. [Офф рекомендация](https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_a_general_checkpoint.html) использовать `.tar`. `pth` не рекомендуется, тк используется в  [питоновских path configuration файлах](https://docs.python.org/3.8/library/site.html)
